src/components/ColorPicker/ColorPicker.tsx

```diff
@@ -0,0 +1,333 @@
+    const rgba = { r, g, b, a: alpha };
+    const hex = toState(rgba).hex;
+    const value = alpha === 1 ? hex : `rgba(${Object.values(rgba).join(", ")})`;
+
+    return { name, rgba, alpha, hex, value };
+};
+
+const ColorPointer: FC = () => <div className="tw-w-4 tw-h-4 tw-rounded-full tw-border-2 tw-border-white" />;
+
+enum ColorFormat {
+    Hex = "Hex",
+    Rgba = "Rgba",
+}
+
+const CustomColorPicker: FC<Omit<ColorPickerProps, "palette">> = ({ currentColor, onSelect }) => {
+    const colorFormats = Object.values(ColorFormat).map((id) => ({ id, title: id.toLocaleUpperCase() }));
+    const [colorFormat, setColorFormat] = useState(ColorFormat.Hex);
+    const [{ hsl, hsv, rgb, hex }, setColor] = useState(tranfsormColor(currentColor));
+    useEffect(() => {
+        setColor(tranfsormColor(currentColor));
+    }, [currentColor]);
+
+    return (
+        <div className="tw-flex tw-flex-col tw-gap-5">
+            <div className="tw-flex tw-gap-2 tw-w-full">
+                <div className="tw-relative tw-flex-grow tw-overflow-hidden tw-rounded">
+                    <Saturation
+                        hsl={hsl}
+                        hsv={hsv}
+                        pointer={() => <ColorPointer />}
+                        onChange={debounce((color) => onSelect(toColor(currentColor, { hex: toState(color).hex })))}
+                    />
+                </div>
+                <div className="tw-relative tw-w-6 tw-overflow-hidden tw-rounded">
+                    <HuePicker
+                        pointer={() => (
+                            <div className="tw-w-6 tw-flex tw-justify-center">
+                                <ColorPointer />
+                            </div>
+                        )}
+                        color={rgb}
+                        direction="vertical"
+                        width="24px"
+                        height="200px"
+                        onChange={debounce(({ hex }) => onSelect(toColor(currentColor, { hex })))}
+                    />
+                </div>
+                <div className="tw-relative tw-w-6 tw-overflow-hidden tw-rounded">
+                    <Alpha
+                        rgb={rgb}
+                        hsl={hsl}
+                        hsv={hsv}
+                        direction="vertical"
+                        width="24px"
+                        height="200px"
+                        pointer={() => (
+                            <div className="tw-w-[18px] tw-flex tw-justify-center">
+                                <ColorPointer />
+                            </div>
+                        )}
+                        style={{ pointer: { top: `${rgb.a * 100}%` } }}
+                        onChange={debounce(({ a }) => onSelect(toColor(currentColor, { rgba: { a } })))}
+                    />
+                </div>
+            </div>
+            <div className="tw-flex tw-gap-2 tw-max-w-full">
+                <div>
```

Comment:
Consider adding a fixed width to avoid jumping of the subsequent elements when changing the color format